.github/workflows/release.yml

```diff
@@ -0,0 +1,103 @@
+name: Build distribution
+on:
+  release:
+    types:
+      - published
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-artifacts:
+    runs-on: ubuntu-latest
+    if: github.repository == 'zarr-developers/VirtualiZarr'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v5
+        name: Install Python
+        with:
+          python-version: "3.12"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install setuptools setuptools-scm wheel twine check-manifest
+
+      - name: Build tarball and wheels
+        run: |
+          git clean -xdf
+          git restore -SW .
+          python -m build --sdist --wheel .
+
+      - name: Check built artifacts
+        run: |
+          python -m twine check dist/*
+          python -m pip check
+          pwd
+          if [ -f dist/virtualizarr-unknown.tar.gz ]; then
+            echo "❌ INVALID VERSION NUMBER"
+            exit 1xxw
+          else
+            echo "✅ Looks good"
+          fi
+      - uses: actions/upload-artifact@v4
+        with:
+          name: releases
+          path: dist
+
+  test-built-dist:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    # environment: # TODO: Have a VirtualiZarr maintainer create a test-release environment
```

Comment:
What needs to be in a `test-release` environment? Are the unit test going to be run in there?

Reply:
Okay. Is this different from a conda environment?

Reply:
Yeah, it's a [GitHub Actions](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) environment. Thanks for sorting out the permissions, I'll work on getting it set up